ddapm_test_agent/trace.py

```diff
@@ -232,11 +232,30 @@ def copy_trace(t: Trace) -> Trace:
 
 def root_span(t: Trace) -> Span:
     """Return the root span of the trace."""
+    # Follow approach used in Datadog Agent: https://github.com/DataDog/datadog-agent/blob/927f9ca9acf7983b72a4bfbdd7a69132e1da8501/pkg/trace/traceutil/trace.go#L53
+
+    if len(t) == 0:
+        raise ValueError("empty trace: %s" % t)
+
+    # common case optimization to check for span where parent_id is either not
+    # set or set to 0
     for s in t:
         if "parent_id" not in s or s["parent_id"] is None or s["parent_id"] == 0:
             return s
 
-    raise ValueError("root span not found in trace: %s" % t)
+    # collect root spans as those with parents that are not themselves spans in trace
```

Comment:
with partial flushing, are we able to have multiple "root" spans where the parent is not in the trace?

```
| ------------------ A1  ----------------- |
| --- B1 --- | | --- B2 --- | | --- B3 --- |
  | -- C1 -- |   | -- C2 -- |   | -- C3 -- |
```

If we take the scenario where partial flushing is enabled with a min span value of `4`.

That would mean when B2 finishes, then B1, C1, B2, C2 would get flushed together in a payload. Then you have multiple "root" spans where the parent is not in the trace.

I know this is likely mostly a very specific edge case, but still possible.

Reply:
I suppose this would be true with a partial flush min span value of 3, 4, 5, or 6.

Reply:
we could quite easily add a test case for this scenario

Reply:
I think today apm assumes that "trace-level" data has to be on the first span of the payload in the event that there are multiple "roots" in the payload.

Reply:
Created an issue to track the work as I don't think it needs to block @majorgreys or this PR: https://github.com/DataDog/dd-apm-test-agent/issues/78